### PR TITLE
daemon: move the last api_foo_test.go to daemon_test

### PR DESCRIPTION
**Pull request by pedronis**
still todo later: clean up what is left in api.go and api_test.go

there are other non api_ tests also still in daemon

### CODE REVIEW

File: daemon/api_console_conf_test.go [hunk 1]
```diff
@@ -17,44 +17,45 @@
  *
  */
 
-package daemon
+package daemon_test
```
Comment (zyga) [inline daemon/api_console_conf_test.go:20]: Woot :)

File: daemon/api_console_conf_test.go [hunk 2]
```diff
@@ -17,44 +17,45 @@
...
 
-	st := d.overlord.State()
+	st := d.Overlord().State()
 
 	body := bytes.NewBuffer(nil)
 	req, err := http.NewRequest("POST", "/v2/internal/console-conf-start", body)
 	c.Assert(err, IsNil)
 
 	// no changes in state, no changes in response
-	rsp := consoleConfStartRoutine(routineConsoleConfStartCmd, req, nil).(*resp)
```
Comment (zyga) [inline daemon/api_console_conf_test.go:55]: It's nice that this part is automatic now.

File: daemon/api_debug_test.go [hunk 3]
```diff
@@ -34,25 +35,27 @@ import (
...
-	APIBaseSuite
+	apiBaseSuite
 }
 
 func (s *postDebugSuite) TestPostDebugEnsureStateSoon(c *check.C) {
 	s.daemonWithOverlordMock(c)
 
 	soon := 0
-	ensureStateSoon = func(st *state.State) {
+	var origEnsureStateSoon func(*state.State)
```
Comment (zyga) [inline daemon/api_debug_test.go:45]: Since the assignment below uses `:=`, what is the point of declaring this explicitly?

Comment (zyga) [inline daemon/api_debug_test.go:45]: Is this defines so that it can be called in the anonymous function below?

**Review by zyga — APPROVED** [review]
Looks good, one question inline.

File: daemon/api_debug_test.go (hunk 3 above)
Comment (pedronis) [reply daemon/api_debug_test.go:45]: yes

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**